src/main/java/edu/ucsb/cs156/happiercows/controllers/CowDeathController.java

```diff
@@ -0,0 +1,91 @@
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Api(description = "Cow Deaths")
+@RequestMapping("/api/cowdeath")
+@RestController
+@Slf4j
+
+public class CowDeathController extends ApiController {
+    @Autowired
+    CommonsRepository commonsRepository;
+
+    @Autowired
+    UserCommonsRepository userCommonsRepository;
+
+    @Autowired
+    CowDeathRepository cowDeathRepository;
+
+    @ApiOperation(value = "Create a new CowDeath as admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/admin/post")
+    public CowDeath postCowDeath_admin(
+            @ApiParam("commonsId") @RequestParam long commonsId,
+            @ApiParam("ZonedDateTime") @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime ZonedDateTime,
+            @ApiParam("cowsKilled") @RequestParam Integer cowsKilled,
+            @ApiParam("avgHealth") @RequestParam Long avgHealth) {
+        
+        Long userId = getCurrentUser().getUser().getId();
+
+        CowDeath createdCowDeath = CowDeath.builder()
+            .commonsId(commonsId)
+            .userId(userId)
+            .ZonedDateTime(ZonedDateTime)
+            .cowsKilled(cowsKilled)
+            .avgHealth(avgHealth)
+            .build();
+        
+        CowDeath savedCowDeath = cowDeathRepository.save(createdCowDeath);
+        return savedCowDeath;
+    }
+
+    @ApiOperation(value = "Get all cow deaths belonging to a commons as an admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @GetMapping("/admin/all/bycommons")
+    public Iterable<CowDeath> allCowDeathsByCommonsId_admin(
+            @ApiParam("commonsId") @RequestParam Long commonsId) {
+        Iterable<CowDeath> cowDeaths = cowDeathRepository.findAllByCommonsId(commonsId);
+        return cowDeaths;
+    }
+
+    
+    @ApiOperation(value = "Get all cow deaths belonging to a user commons as a user via CommonsID")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all/byusercommons")
+    public Iterable<CowDeath> allCowDeathsByCommonsId(
+            @ApiParam("commonsId") @RequestParam Long commonsId) {
+        
+        Long userId = getCurrentUser().getUser().getId();
+        UserCommons userCommons = userCommonsRepository.findByCommonsIdAndUserId(commonsId, userId)
+            .orElseThrow(() -> new EntityNotFoundException(UserCommons.class, "commonsId", commonsId, "userId", userId));
+
+        if (userId != userCommons.getUserId())
+            throw new EntityNotFoundException(UserCommons.class, userCommons.getId());
+
+        Iterable<CowDeath> cowDeaths = cowDeathRepository.findAllByCommonsId(userCommons.getId());
```

Comment:
I don't think this is correct ( and when I tested it, it did not work).

I think the issue that that you are doing a look up that expects a commonsId, but you are passing a userCommonsId.

I wonder if the correct  code should be something like this:

```suggestion
Iterable<CowDeath> cowDeaths = cowDeathRepository.findAllByCommonsId(userCommons.getCommonsId());
```

But even that is not correct, because we only want the CowDeath records that are for this particular user.

Does that make sense?  The UserCommons is an entity that tracks that use Alice is part of Common "Irene's Commons", or that Bob is part of Common "Irene's Commons".  

When Alice sends a request to this endpoint and passes in the CommonsId for Irene's commons, we only want to see CowDeaths that pertain to Alice, not the ones that pertain to Bob.

Reply:
Bottom line: you may need to add a new query type to the `CowDeathRepository` that looks things up by user id and commons id.

Reply:
Fixed